docs/synchronization.md

```diff
@@ -52,7 +53,7 @@ spec:
       args: ["hello world"]
 ```
 
-Using a mutex:
+Using a mutex achieves the same thing as a count `"1"` semaphore:
```

Comment:
Can we add an example where the `mutex` name is dynamic?

For example, a workflow or multiple workflows that interact with the same external resource such as a database would want to prevent copy/backup/restore workflows from occurring at the same time.

```
mutex:
   name: "database-lock-{{inputs.parameters.dbName}}"
```

- database-backup workflow A  - locks database A
- database-copy workflow B - waits for database-backup workflow A to release the mutex

Reply:
We could. It's outside the scope of what I was trying to achieve in this PR, and would prefer someone did that in a separate PR.